File: Modules/Common/Equipments/SIE.Web.Equipments/Abnormal/Commands/AbnormalCauseAddCommand.js
/*
 ** 异常停线添加命令，初始化时间
 * @class SIE.Web.Equipments.Abnormal.Commands.AbnormalCauseAddCommand
 */
SIE.defineCommand('SIE.Web.Equipments.Abnormal.Commands.AbnormalCauseAddCommand', {
    extend: 'SIE.cmd.Add',
    meta: { text: "添加", group: "edit", iconCls: "icon-AddEntity icon-green" },
    canExecute: function (view) {
        return true;
    },
    getEditEntity: function () {
        var newEntity = Ext.create(this.view.model);
        if (this.view.isListView) {
            newEntity = this.createNewItem();
        }
        //如果停线来源是“自建”，停线类别为null
        if (newEntity.getSourceType() === "0") {
            newEntity.setExceptionStopType("");
        }
        this.onItemCreated(newEntity);
        return newEntity;
    },
    onItemCreated: function (entity) {
        if (entity) {
            var model = entity.data;
            var me = this;
            this.view.execute({
                data: model,
                success: function (res) {
                    var data = res.Result;
                    entity.setBeginDate(data.BeginDate);
                    entity.setCode(data.Code);
                }
            }, me.view);
        }
    }
});

File: Modules/Common/Equipments/SIE.Web.Equipments/Abnormal/Commands/AbnormalCauseAddCommand.cs
using SIE.Equipments.Abnormal;
using SIE.Web.Command;
using System;

namespace SIE.Web.Equipments.Abnormal.Commands
{
    /// <summary>
    /// 异常停线添加命令
    /// </summary>
    public class AbnormalCauseAddCommand : ViewCommand
    {
        /// <summary>
        /// 初始化日期时间
        /// </summary>
        /// <param name="args">参数</param>
        /// <param name="scope">作用域</param>
        /// <returns>实体</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var abnormalCause = args.Data.ToJsonObject<AbnormalCause>();
            abnormalCause.BeginDate = DateTime.Now;
            abnormalCause.Code = AppRuntime.Service.Resolve<AbnormalCauseController>().GetNewAbnormalCode();
            return abnormalCause;
        }
    }
}
